app/src/stores/files.ts

```diff
@@ -0,0 +1,10 @@
+import { defineStore } from 'pinia';
+import { ref } from 'vue';
+
+export const useFilesStore = defineStore('filesStore', () => {
+	const uploading = ref(false);
```

Comment:
When we're relying on a global true/false state, it'll mean that different consumers of this store can set it to false even though other processes are still uploading.

Instead, the internal state of the files store should be an array of upload IDs. We can export a `startUpload` function that itself returns the corresponding stop function. That way the upload component(s) can consume it as:

```js
const filesStore = useFilesStore();

const done = filesStore.startUpload();

// Do some async work to upload the file(s)

done();
```

Internally, this store could look something like (pseudo code):

```js
const active = ref([]);
const uploading = computed(() => active.value.length > 0);

const start = () => {
  const id = genId();
  active.value.push(id);

  return () => {
    active.value = active.value.filter((act) => act !== id));
  }
}

return { uploading, start };
```